pkg/autodiscover/autodiscover_operators.go

```diff
@@ -63,32 +64,47 @@ func isIstioServiceMeshInstalled(allNs []string) bool {
 	return true
 }
 
-func findOperatorsByLabel(olmClient clientOlm.Interface, labels []labelObject, namespaces []configuration.Namespace) []*olmv1Alpha.ClusterServiceVersion {
-	csvs := []*olmv1Alpha.ClusterServiceVersion{}
+func findOperatorsByLabel(olmClient clientOlm.Interface, labels []labelObject, namespaces []configuration.Namespace) (csvs []*olmv1Alpha.ClusterServiceVersion, err error) {
+	csvs = []*olmv1Alpha.ClusterServiceVersion{}
+	var csvList *olmv1Alpha.ClusterServiceVersionList
+	labelSelector := ""
+
+	// If labels are provided, create the label selector
+	if len(labels) > 0 {
+		labelSelector = labels[0].LabelKey + "=" + labels[0].LabelValue
+		for _, l := range labels[1:] {
+			labelSelector += "," + l.LabelKey + "=" + l.LabelValue
+		}
+	}
```

Comment:
I feel like this could be in its own function with corresponding unit tests to see what comes out the other side.